src/FilenameParsing/FileIDHelper.php

```diff
@@ -53,4 +53,11 @@ public function isVariantOf($fileID, ParsedFileID $parsedFileID);
      * @return string
      */
     public function lookForVariantIn(ParsedFileID $parsedFileID);
+
+    /**
+     * Specify if this File ID Helper stores variants in subfolders and require a recursive look up to find all
+     * variants.
+     * @return bool
+     */
+    public function lookForVariantRecursive(): bool;
```

Comment:
Technically this is a breaking API change because a class implementing this interface will get an error if it doesn't implement this method. However this interface is marked as `internal` so we're allowed do to breaking changes to it.

Reply:
In practical terms I wouldn't expect there to be any custom implementations of this out in the wild, so this is probably OK

Reply:
Could do it non-breaking by just adding this function to the classes themselves instead of the interface and doing a method_exists() check before calling, and if it doesn't exist, default to true

Reply:
We could, but I expect that at some stage we'll make this interface part of our official API. I would rather have a nice clean API when we do that. The entire point of marking these classes as internal was that we could do something like this.

Reply:
Oh right gotcha, sorry I didn't realise all these classes were marked internal.